src/Microsoft.AspNet.SignalR.JS/jquery.signalR.core.js

```diff
@@ -979,7 +979,6 @@
             }
 
             // Always clean up private non-timeout based state.
-            delete connection._.config;
```

Comment:
Could we instead move the `(!connection._.config || connection._.config.waitForPageLoad === true)` code and assign it to a local variable before deleting config? That way we don't have to make sure we delete config in all conditions below.

Reply:
Well, good idea